src/main/webapp/mapsScript.js

```diff
@@ -90,6 +93,93 @@ function showDirections(locations) {
       window.alert('Directions request failed due to ' + status);
     }
   });
+
+  // Add markers to map.
+  addPoiMarker(origin, true, 0);  
+  for (let i = 0; i < waypts.length; i++) {
+    let index = i + 1;
+    addPoiMarker(waypts[i].location, false, index);
+  }
+}
+
+/* 
+ * Adds marker to map with appropriate letter label and an info window.
+ * location: String name or address of the POI.
+ * isOrigin: boolean indicating if the location is the origin (hotel) or not.
+             Used to determine the icon of the map marker.
+ * index: index of waypoint. Used to determine marker label.
+ */
+function addPoiMarker(location, isOrigin, index) {
+  // Query the location String to get Place details.
+  var service = new google.maps.places.PlacesService(map);
+  service.textSearch({
+    location: map.getCenter(),
+    radius: '500',
+    query: location
+    }, function(results, status) {
+      if (status == google.maps.places.PlacesServiceStatus.OK) {
+        // Add marker to map.
+        let marker = new google.maps.Marker({
+          map: map,
+          place: {
+            placeId: results[0].place_id,
+            location: results[0].geometry.location
+          },
+        });
+        // Change icon to green if the location is the origin.
+        if (isOrigin) {
+          marker.setIcon("http://maps.google.com/mapfiles/ms/icons/green-dot.png");
+        } else {
+          marker.setLabel({
+            color: "white",
+            // Set text to correct letter label.
+            text: labels[index % labels.length]
+          });
+        } 
+        // Add info window to marker.     
+        addInfoWindow(results, marker);
+      } else {
+        alert(status);
+        return;
+      }
+  });
+}
+
+/* 
+ * Adds info window to a marker.
+ * Info window includes place name, address, and link to Google Maps.
+ */
+function addInfoWindow(results, marker) {
+  // Format and add place name, address, and link to Google Maps.
+  let contentString = "";
+  contentString += "<b>" + results[0].name + "</b>";
+  contentString += "<p>" + results[0].formatted_address + "</p>";
+  const urlFormatAddress = results[0].name.replace(/\s/g, '+');
+  const url = 'https://www.google.com/maps/search/?api=1&query=' + urlFormatAddress 
+              + '&query_place_id=' + results[0].placeId;
+  contentString += "<a href=\""+url+"\" target=\"_blank\">View on Google Maps</a>";
```

Comment:
If you're creating an HTML object, you should probably do it dynamically in Javascript. `document.createElement('p')` etc... You can look at my PopUp branch, I changed mine as well